open_humans/templates/partials/visible-public-sharing.html

```diff
@@ -1,7 +1,5 @@
 <div class="">
-  {% if data_files %}
-
-    {% if user.member.public_data_participant.enrolled %}
+    {% if not project_member.revoked %}
```

Comment:
I think this should be switched ... probably to also check `project_member.authorized` as done for the `render_user_badges` templatetag below.

Because `revoked` defaults to false on membership objects I'd worry a bit about accidentally having membership objects interpreted as being joined/authorized when they just failed to set `revoked` to True.

Regarding this whole page & inconsistency: `member/activity-management.html` template is checking `activity.is_connected`, which is determined by `personalize_activities_dict` in `common/activities.py`, which generally decides based on the `project.is_joined` boolean.

The `joined` and `authorized` distinction is an odd one that reflects a two-stage process for on-site activities that present a consent/terms = "joined", followed by an authorization = "authorized". In OAuth2 projects these should be synonymous. Since `authorized` is the second step, I think it's the one to we should use.

**All the code in `common/activities.py` is probably legacy.** (I'm taking this opportunity to make this point. :slightly_smiling_face: ) We used that stuff to combine "app-style projects" (legacy) with "model-style projects". The app style projects are gone now, the only non-standard thing that remains is "public data sharing".

Reply:
Hrm, at this point, it'll probably be easier to just make a templatetag that outputs the html rather than having twenty nested ifs.  Instead, I'll just do a nice complex db query like I've started doing elsewhere.  That'll be easier in the templatetag than in the template itself.

As for common.activities, I sort of gathered that, but that'll be an involved project in itself to rip all that out.  There's still a few things that depend on that code that could be better done as direct db queries.

Reply:
with the templates - agreed complex logic is rough! my understanding is that moving it into views is usually ideal (rather than templatetags).

and... I still think this line should change, even if it's a minor change :)